processor/resourcedetectionprocessor/README.md

```diff
@@ -318,14 +318,18 @@ processors:
 
     * cloud.provider ("aws")
     * cloud.platform ("aws_eks")
+    * k8s.cluster.name
+
+Note: The kubernetes cluster name is only available when running on EC2 instances, and requires permission to run the `EC2:DescribeInstances` [action](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstances.html).
```

Comment:
We can also check if tags are available on the IMDS endpoint first. So if they are exposed on imds, we don't have to make the describe instance call. EC2 instances have a [setting to expose tags through IMDS](https://aws.amazon.com/about-aws/whats-new/2022/01/instance-tags-amazon-ec2-instance-metadata-service/).

Reply:
I know it's possible, but it seemed more complicated to have two routes here. In my test environment `DescribeInstances` was allows, but `DescribeTags` wasn't by default, so this solution was what worked for me. 

Your suggestion would help in the case where permission is granted to run `DescribeTags` but not `DescribeInstances`. Do you think there's some level of likelihood that this would happen? If it is I'm happy to add the check, but I'd prefer to keep it simple if it's not something we expect to run into.

Reply:
Right, but this is trying to access tags from the local instance metadata service which shouldn't require an IAM role, just access to the local imds endpoint (the same way we fetched the instance id). But nevermind, the tags are not available on the imds endpoint by default and requires an additional config to be set in the ec2 instances.